app/models/generic_object.rb

```diff
@@ -95,6 +95,22 @@ def property_setter(name, value)
     self.properties = properties.merge(name => val)
   end
 
-  def call_automate(method_name, options = {}, q_options = {})
+  # the method parameters are passed into automate as an array in $evm.root['args']
+  # the return value from automate is in $evm.root['method_result']
+  def call_automate(method_name, *args)
+    options = {
+      :object_type   => self.class.name,
+      :object_id     => id,
+      :instance_name => 'GenericObject',
+      :user_id       => User.current_user.id,
+      :attrs         => {
+        :go_definition => generic_object_definition.name,
+        :method_name   => method_name,
+        :args          => args,
```

Comment:
The `args` variable needs to be a hash and it gets merged into `:attrs`.

To prevent overriding the values we are using to build the relationship you can use `reverse_merge`.  Also suggest using `go_definition_name`:

```ruby
:attrs => args.reverse_merge(:go_definition_name => generic_object_definition.name, 
                             :method_name        => method_name)
```